server/database.js

```diff
@@ -223,9 +223,31 @@ class Database {
 
         let config = {};
 
+        const DEFAULT_MAX_POOL_CONNECTIONS = 10;
+        const maxPoolConnections = process.env.UPTIME_KUMA_DB_POOL_MAX_CONNECTIONS;
+        let parsedMaxPoolConnections = parseInt(maxPoolConnections);
+
+        if (
+            !maxPoolConnections ||
+            maxPoolConnections.trim() === "" ||
+            Number.isNaN(parsedMaxPoolConnections) ||
+            parsedMaxPoolConnections > 100 ||
+            // NOTE: We cap the pool size because MySQL/MariaDB connections are heavy.
+            // Each connection consumes memory and CPU, and too many cause severe slowdown.
+            // If you need higher concurrency, use a proxy like ProxySQL or MaxScale instead of raising this limit.
+            parsedMaxPoolConnections < 1
+        ) {
+            parsedMaxPoolConnections = DEFAULT_MAX_POOL_CONNECTIONS;
+
+            // Only log if there was a UPTIME_KUMA_DB_POOL_MAX_CONNECTIONS defined
+            if (maxPoolConnections !== undefined) {
+                log.warn("db", `Max pool connections defaulted to ${DEFAULT_MAX_POOL_CONNECTIONS} because UPTIME_KUMA_DB_POOL_MAX_CONNECTIONS was invalid`);
```

Comment:
Plets split the cases to improve the warnings. "being invalid" is very unspecific.